apps/balancer-tools/src/app/apr/(utils)/calculatePoolStats.ts

```diff
@@ -38,17 +39,17 @@ export async function fetchPoolAveragesForDateRange(
   const extendedFrom = initialRangeInDays < 2 ? from - SECONDS_IN_DAY : from;
 
   // Fetch snapshots within the (potentially extended) date range
-  const res = await pools.gql(network).poolSnapshotInRange({
-    poolId,
-    timestamp: [extendedFrom],
-  });
+  const res = await fetchPoolSnapshots(to, extendedFrom, network, poolId);
 
   let chosenData = res.poolSnapshots[0];
 
   if (res.poolSnapshots.length == 0) {
     const retryGQL = await pools.gql(network).poolSnapshotInRange({
       poolId,
-      timestamp: generateDateRange(from - SECONDS_IN_DAY * 7, to),
+      timestamp: generateDateRange(
+        from - SECONDS_IN_DAY * 7,
+        to - SECONDS_IN_DAY,
```

Comment:
If the original `to` already failed then there is no need to try it again

Reply:
IMO there's no need to add this complication though